django_vaultkeeper_adaptor/vaultkeeper_adaptor.py

```diff
@@ -13,20 +14,27 @@ def build_ids_url(backend, endpoint, identifier, secret):
 class VKAdaptor(object):
 
     def __init__(self,
-                 data=None,
+                 config_path=None,
                  DATABASES=None,
                  BROKER_URL=None,
                  ):
+        self.config_path = config_path
         self.DATABASES = DATABASES
         self.BROKER_URL = BROKER_URL
+        self.secrets = {}
 
         self.functionmap = {
             'postgresql': self.get_database_configs,
             'rabbitmq': self.get_broker_url,
         }
 
-        self.data = data
-        self.secrets = {}
+        try:
+            with open(self.config_path) as f:
+                data = json.load(f)
+                self.data = data
+        except IOError:
+            raise IOError("Error: File {} not found.".format(
+                          self.config_path))
```

Comment:
The default IOError message here is something like `IOError: [Errno 2] No such file or directory: '{filename}'`. I'm not sure that this error message is sufficiently different to warrant catching the error and raising one of the exact same type.

Also, there are different kinds of IOErrors, like `IOError: [Errno 13] Permission denied`, which in this case will print out with the above file not found message, which might be confusing.